### ISSUE TITLE
Affinity routing uses global np.random without seeding; benchmark not reproducible

### ISSUE THREAD

**Issue by sachncs** [sachncs/tsn-affinity#10]
## Problem

Affinity routing uses NumPy global RNG (`np.random.randint`) without seeding, so the batches drawn for action-affinity estimation differ on every run. Two consecutive calls to `tsn-benchmark` with identical configuration produce different affinity scores, different routing decisions, and different copy-creation outcomes — defeating the "Reproducible benchmarking with ... Statistical validation across multiple runs" claim made in `tsn_affinity/cli/benchmark.py:8`.

## Evidence

`tsn_affinity/routing/metrics.py:139` (`_MultiBatchLoader._generate_batches`):

```python
for _ in range(batch_size):
    traj_idx = np.random.randint(0, n_trajs)
    ...
    start = 0 if T <= seq_len else np.random.randint(0, T - seq_len + 1)
```

`tsn_affinity/routing/metrics.py:510` (`_AffinityBatchLoader._build_batches`):

```python
traj_idx = np.random.randint(0, n_trajs)
...
start = np.random.randint(0, T - seq_len + 1)
```

`tsn_affinity/data/loaders/batch_loader.py:45,51` (`make_minibatches`):

```python
traj = trajs[np.random.randint(len(trajs))]
...
start = np.random.randint(0, T - seq_len + 1)
```

`tsn_affinity/data/loaders/panda_loader.py:56,62` — same pattern.

Only `tsn_affinity/cli/benchmark.py:372-373` and `run_single_benchmark` (line 385) call `np.random.seed` / `torch.manual_seed`, but the affinity batches are generated *after* that point by `compute_action_affinity_batch`, which spawns its own `_MultiBatchLoader` and uses the global NumPy RNG.

`run_single_benchmark` calls `np.random.seed(config.seed + seed_offset + task_id * 1000)` (line 385) for *trajectory generation*, then affinity uses the global RNG (which has been advanced by `make_minibatches` calls during training).

## Root Cause

The CLI seeds NumPy once before data generation but does not re-seed before affinity computation, and the affinity code path uses the global RNG.

## Impact

- `README.md:147-151` reports ACC/BWT stddevs of `±0.0028`–`±0.0041` across 3 runs. With affinity-batch RNG uncontrolled, the stddev captures both legitimate run-to-run variance and noise from affinity-batch sampling. The published numbers cannot be reproduced bit-for-bit by a third party.
- `tsn_affinity/strategies/tsn_affinity.py:_select_copy_for_new_task` consumes affinity scores; if the same task is evaluated twice with different RNG draws, the routing decision can flip between reusing a copy and creating a new one.
- Reproducibility claim in the CHANGELOG (`CHANGELOG.md:88-91` mentions "Reproducible benchmarking") is partially violated.

## Recommendation

1. Replace `np.random.randint` calls in `tsn_affinity/routing/metrics.py` and the data loaders with a per-call `np.random.default_rng(seed)` instance seeded from the strategy/task IDs (mirroring how `_build_task_memory` already seeds in `tsn_affinity/strategies/tsn_affinity.py:137`).
2. Document the determinism contract per-affinity-batch in the docstring.
3. Add a test that calls `compute_action_affinity_batch` twice with the same seed and asserts identical scores.

## Acceptance Criteria

- [ ] Two calls to `compute_action_affinity_batch(models, trajs, ..., device="cpu")` with the same inputs produce identical affinity dictionaries
- [ ] Two consecutive `tsn-benchmark` runs with the same `--seed` produce identical `runs/.../performance_matrix.npy` files
- [ ] The docstring on `_MultiBatchLoader` documents the seed contract

## Verification

```
python -c "
from tsn_affinity.routing.metrics import compute_action_affinity
import torch
from tsn_affinity.core.decision_transformer import DecisionTransformer
m = DecisionTransformer(obs_shape=(4,), n_actions=2, d_model=16, n_layers=1, n_heads=2)
trajs = ... # some trajectories
torch.manual_seed(0)
s1 = compute_action_affinity(m, trajs, seq_len=5, n_batches=2, batch_size=4, device='cpu')
torch.manual_seed(0)
s2 = compute_action_affinity(m, trajs, seq_len=5, n_batches=2, batch_size=4, device='cpu')
assert s1 == s2
"
```

## Risk

Low. Same algorithm, just deterministic. May slightly change published numbers.

**Comment by sachncs** [comment]
Fixed in commit a799277. BaseStrategy now accepts a seed argument and seeds numpy + torch. RoutingConfig carries a seed field that propagates into the warm-starter and the strategy-local generator. The affinity router's np.random.default_rng calls (used for task memory sampling) now use the per-strategy generator instead of the global numpy state.
